ProcessMaker/Services/ScriptMicroserviceService.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace ProcessMaker\Services;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use ProcessMaker\Events\ScriptResponseEvent;
+use ProcessMaker\Jobs\CompleteActivity;
+use ProcessMaker\Models\Process as Definitions;
+use ProcessMaker\Models\ProcessRequest;
+use ProcessMaker\Models\ProcessRequestToken;
+use ProcessMaker\Models\User;
+
+class ScriptMicroserviceService
+{
+    public function handle(Request $request)
+    {
+        $response = $request->all();
+        Log::debug('Response microservice executor: ' . print_r($response, true));
+        // If the call is from preview
+        if ($response['metadata']['nonce']) {
+            $status = $response['status'] === 'success' ? 200 : 500;
+            $output = $response['status'] === 'success'
+                ? ['output' => $response['output']]
+                : $response;
+
+            event(new ScriptResponseEvent(
+                User::find($response['metadata']['user_id']),
```

Comment:
@gusys Could the `User::find()` return null?

Reply:
This is coming from a 100% UI (Scripts IDE) so the user will be always present. @sanjacornelius